app/src/main/java/com/example/houta/dadvice_v01/CardAdapter.java

```diff
@@ -0,0 +1,65 @@
+package com.example.houta.dadvice_v01;
+
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+
+/**
+ * Created by houta on 10/15/2016.
+ */
+// template: https://developer.android.com/training/material/lists-cards.html
+// guidance: http://icetea09.com/blog/2014/12/19/android-cardview-and-recyclerview-in-material-design/
+public class CardAdapter extends RecyclerView.Adapter<CardAdapter.ViewHolder> {
+    private ArrayList<String> mDataset;
+
+    // provide a reference to the views for each data item
+    // complex data items may need more than one view per item
+    // you provide access to all the views for a data item in a view holder
+
+    public static class ViewHolder extends RecyclerView.ViewHolder {
+        // each data item is just a string in this case
+        public TextView mTextView;
+        // viewHolder constructor
+        // ref: http://stackoverflow.com/questions/27450598/android-widget-textview-cannot-be-applied-to-android-view-view
+        // ref: http://stackoverflow.com/questions/28407768/android-widget-linearlayout-cannot-be-cast-to-android-widget-textview-in-recycle
+        public ViewHolder(View v){
+            super(v);
+            mTextView = (TextView) v.findViewById(R.id.textView);
+        }
+    }
+
+    // provide a suitable constructor (depends on kind of dataset)
+    public CardAdapter(ArrayList<String> myDataset) {
+        mDataset = myDataset;
```

Comment:
Clever but simple constructor!  In the future we can add error checking/maximum bounds here, not a priority for a long time though